ui-src/src/components/Minimap.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import './Minimap.css';
+
+import { connect } from 'react-redux';
+
+import * as common from '../common';
+import store from '../store';
+import {StoreState} from '../types';
+
+
+const MAX_MINIMAP_SIZE = 200
+
+type MinimapProps = {}
+type MinimapState = {
+  swapState: boolean
+}
+
+type MinimapViewport = {
+  x: number,
+  y: number,
+  w: number,
+  h: number,
+}
+
+class Minimap extends React.Component<MinimapProps, MinimapState> {
+
+  private viewport: MinimapViewport = {
+    x: 0, y: 0, w: 0, h: 0
+  }
+  private canvas = React.createRef<HTMLCanvasElement>()
+  private drawInterval: any = null
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      swapState: false
+    }
+  }
+
+  public setViewport(viewport: MinimapViewport) {
+    this.viewport = viewport
+    this.forceUpdate()
+  }
+
+  public componentWillMount() {
+    const {gameBoard} = store.getState().currentGame!
+    const {x, y} = gameBoard.size
+  }
+
+  public componentDidMount () {
+    this.draw()
+    this.drawInterval = setInterval(this.draw, common.MINIMAP_DRAW_INTERVAL)
+  }
+
+  public componentWillUnmount() {
+    clearInterval(this.drawInterval)
+  }
+
+  public render() {
+    const {gameBoard} = store.getState().currentGame!
+    const [[width, height], scale] = this.sizeAndScale()
+    const {x, y, w, h} = this.viewport
+    const className = this.state.swapState ? 'swap' : ''
+    const viewportStyle = {
+      left: (x * 100) + '%',
+      top: (y * 100) + '%',
+      width: (w * 100) + '%',
+      height: (h * 100) + '%',
+      display: w >= 1 || h >= 1 ? 'none' : 'block',
+    }
+    return <div className={"minimap " + className} onMouseOver={this.onMouseOver} >
```

Comment:
Good addition!!  I like the idea of moving the map (instead of changing the transparency) to allow players to select the cells behind the minimap.